File: src/internal/entities/exchange_rate.go
package entities

type ExchangeRatePrimitive struct {
	RequestResult           int    `json:"result"`
	CurrencyCode            string `json:"cur_unit"`
	CurrencyName            string `json:"cur_nm"`
	BuyingRate              string `json:"ttb"`
	SellingRate             string `json:"tts"`
	ExchangeRate            string `json:"deal_bas_r"`
	BookPrice               string `json:"bkpr"`
	YearHigh                string `json:"yy_efee_r"`
	TenDaysHigh             string `json:"ten_dd_efee_r"`
	MoneyBrokerExchangeRate string `json:"kftc_deal_bas_r"`
	MoneyBrokerBookPrice    string `json:"yy_kftc_bkpr"`
}

type ExchangeRate struct {
	CurrencyCode string  `json:"currency_code"`
	CurrencyName string  `json:"currency_name"`
	Date         string  `json:"date"`
	DirectRate   float64 `json:"direct_rate"`
}

File: app/lib/internal/entities/currency/exchange_rate.dart
final class ExchangeRate {
  const ExchangeRate({
    required this.currencyCode,
    required this.currencyName,
    required this.date,
    required this.directRate,
  });

  final String currencyCode;
  final String currencyName;
  final String date;
  final double directRate;

  factory ExchangeRate.fromMap(Map<String, dynamic> map) {
    return ExchangeRate(
      currencyCode: map["currency_code"] as String,
      currencyName: map["currency_name"] as String,
      date: map["date"] as String,
      directRate: map["direct_rate"] as double,
    );
  }
}
